modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -243,93 +245,94 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
         self._info("Handled OnBegin Callbacks.")
 
         batch_number = 0
-        for batch_number, batch in enumerate(self._train_dataloader):
-            for _, callback in self._callbacks.items():
-                callback.on_batch_begin(self._model.model, self._optimizers, batch, batch_number)
-
-            # As empty() is unreliable
-            # we try to fetch an element within 10ms. If there is no
-            # element within that timeframe returned, we continue.
-            try:
-                req = self._status_query_queue.get(timeout=0.01)
-                if req == TrainerMessages.STATUS_QUERY_MESSAGE:
-                    self.send_status_to_server(batch_number)
-                elif req == TrainerMessages.MODEL_STATE_QUERY_MESSAGE:
-                    self.send_model_state_to_server()
+        for _ in range(self.epochs_per_trigger):
+            for batch_number, batch in enumerate(self._train_dataloader):
```

Comment:
please add a test that this setting actually works. I don't think it would right now. the OnlineDataset has some internal state that always fetches the next training set, which means this would not always fetch the same data

Reply:
I tried doing simple tests (adding a for loop to existing tests) and trying to run by analysing the chosen keys run by run and it seems to work. Any other clever tests I could add?